packages/@aws-cdk/aws-stepfunctions-tasks/lib/emrcontainers/start-job-run.ts

```diff
@@ -0,0 +1,710 @@
+              LogUri: (this.props.monitoring?.logging === true)
+                ? `s3://${this.renderS3BucketName()}`// automatically generated name https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-s3.Bucket.html#bucketname
+                : `s3://${this.props.monitoring?.logBucket?.bucketName}`,
+            },
+          },
+          Tags: this.props.tags ? this.renderTags(this.props.tags) : undefined,
+        },
+      }),
+    };
+  }
+
+
+  /**
+   * Render the EMR Containers ConfigurationProperty as JSON
+   *
+   * @param property
+   */
+  private ApplicationConfigPropertyToJson(property: ApplicationConfiguration) {
+    return {
+      Classification: cdk.stringToCloudFormation(property.classification.classificationStatement),
+      Properties: cdk.objectToCloudFormation(property.properties),
+      Configurations: cdk.listMapper(this.ApplicationConfigPropertyToJson)(property.nestedConfig),
+    };
+  }
+
+  private renderTags(tags: { [key: string]: any } | undefined): { [key: string]: any } {
+    return tags ? { Tags: Object.keys(tags).map((key) => ({ Key: key, Value: tags[key] })) } : {};
+  }
+
+  /**
+   * Creates a Cloudwatch log group
+   *
+   * @returns name if user enables props.monitoring.logging to be true
+   **/
+  private renderLogGroupName(): string {
+    return new logs.LogGroup(this, 'Emr Containers Default Cloudwatch Log Group').logGroupName;
+  }
+
+  /**
+   * Creates a S3 bucket
+   *
+   * @returns an automatically generated bucket name
+   **/
+  private renderS3BucketName(): string {
+    return new s3.Bucket(this, 'Emr Containers Default S3 Bucket').bucketName;
+  }
+
+  // https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/creating-job-execution-role.html
+  private createJobExecutionRole(): iam.Role {
+    let jobExecutionRole: iam.Role = new iam.Role(this, 'Job-Execution-Role', {
+      assumedBy: new iam.ServicePrincipal('emr-containers.amazonaws.com'),
+    });
+
+    // allow job-execution-role to be used by Step Functions
+    jobExecutionRole.assumeRolePolicy?.addStatements(
+      new iam.PolicyStatement({
+        actions: ['sts:AssumeRole'],
+        principals: [new iam.ServicePrincipal('states.amazonaws.com')],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers Managing Virtual Clusters Policies', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
+            actions: [
+              'emr-containers:CreateVirtualCluster',
+              'emr-containers:DeleteVirtualCluster',
+              'emr-containers:DescribeVirtualCluster',
+              'emr-containers:ListVirtualClusters',
+            ],
+          }),
+        ],
+      }),
+    );
+
+    jobExecutionRole.addManagedPolicy(
+      new iam.ManagedPolicy(this, 'Emr Containers S3 and Log Policies', {
+        statements: [
+          new iam.PolicyStatement({
+            resources: ['*'],
```

Comment:
these should be scoped to the bucket and log groups that are passed in/created (and only added if logging is enabled)

Reply:
Removed because redundant.